app/src/main/java/com/android/ootd/model/items/ItemsRepositoryFirestore.kt

```diff
@@ -76,141 +99,51 @@ class ItemsRepositoryFirestore(private val db: FirebaseFirestore) : ItemsReposit
-        mapToItem(doc)?.let {
-          itemsCache[it.itemUuid] = it
-          matches.add(it)
-        }
-      }
-    }
-    return matches
-  }
-
   override suspend fun addItem(item: Item) {
-    // Optimistically update memory cache immediately
-    itemsCache[item.itemUuid] = item
-    Log.d(TAG, "Added item ${item.itemUuid} to memory cache")
-
-    // Queue Firestore update with timeout
     try {
-      kotlinx.coroutines.withTimeoutOrNull(2_000L) {
+      withTimeout(2000L) {
         db.collection(ITEMS_COLLECTION).document(item.itemUuid).set(item).await()
       }
-      Log.d(TAG, "Item added to Firestore (or queued if offline)")
+    } catch (e: TimeoutCancellationException) {
+      // Timeout means offline; Firestore will sync later
     } catch (e: Exception) {
-      Log.w(TAG, "Firestore add queued (offline): ${e.message}")
+      Log.e("ItemsRepositoryFirestore", "Error adding item: ${e.message}", e)
+      throw e
     }
   }
 
   override suspend fun editItem(itemUUID: String, newItem: Item) {
-    // Determine if item exists (cache fast-path, then Firestore with timeout)
-    val existedInCache = itemsCache.containsKey(itemUUID)
-    var existedInStore = false
-    if (!existedInCache) {
+    try {
+      // Check existence first (optional, but good for consistency)
+      // We use a short timeout for existence check
       try {
-        existedInStore =
-            kotlinx.coroutines.withTimeoutOrNull(1_000L) {
-              db.collection(ITEMS_COLLECTION).document(itemUUID).get().await().exists()
-            } ?: false
-      } catch (e: Exception) {
-        // Offline / timeout: treat as not found in store for contract purposes
-        existedInStore = false
-        Log.w(TAG, "Existence check timed out/offline: ${e.message}")
+        withTimeout(1000L) {
+          val doc = db.collection(ITEMS_COLLECTION).document(itemUUID).get().await()
+          if (!doc.exists()) {
+            throw Exception("ItemsRepositoryFirestore: Item not found")
+          }
+        }
+      } catch (e: TimeoutCancellationException) {
+        // Timeout on existence check; assume exists and try to update
       }
-    }
-    if (!existedInCache && !existedInStore) {
-      throw Exception("ItemsRepositoryFirestore: Item not found")
-    }
-
-    // Optimistically update memory cache immediately (synchronous, instant)
-    itemsCache[itemUUID] = newItem
-    Log.d(TAG, "Updated item $itemUUID in memory cache")
 
-    // Queue Firestore update with timeout (will sync when online, doesn't block)
-    try {
-      kotlinx.coroutines.withTimeoutOrNull(2_000L) {
-        db.collection(ITEMS_COLLECTION).document(itemUUID).set(newItem).await()
-      }
-      Log.d(TAG, "Item edited in Firestore (or queued if offline)")
+      withTimeout(2000L) { db.collection(ITEMS_COLLECTION).document(itemUUID).set(newItem).await() }
+    } catch (e: TimeoutCancellationException) {
+      // Timeout means offline; Firestore will sync later
```

Comment:
So, if I understand correctly that this section is for when we cannot reach the Firestore, so it means that the app is in offline mode. To indicate in the log maybe you can implement an indicator like 

```suggestion
      // Timeout means offline; Firestore will sync later
      Log.i(TAG, "Item is queued for offline mode") 
```

What do you think about it ?